dist/gcc-compatible/Hacl_SHA2_Types.h

```diff
@@ -35,30 +35,17 @@ extern "C" {
 #include "krml/lowstar_endianness.h"
 #include "krml/internal/target.h"
 
-typedef struct Hacl_Hash_SHA2_uint8_2p_s
```

Comment:
I don't know why these typedefs changed in the current extraction. 
It seems like a regression since it results in bad type names in the C code.

Reply:
yeah this is blocking because now sha2 depends on sha3 (which is *not* what we want)

these types are poorly named -- they should be named "generic types" or something but let's leave that aside for now

the way to fix this is to figure out *where* in Hacl_SHA3 pairs of pointers to uint8 are used, and have those pairs be references to Hacl.SHA2.Types instead of un-named types (like `buffer uint8 & buffer uint8`)